src/mozanalysis/sizing.py

```diff
@@ -0,0 +1,358 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from enum import Enum
+
+import attr
+import datetime
+
+from mozanalysis.bq import sanitize_table_name_for_bq
+from mozanalysis.utils import add_days, date_sub, hash_ish
+from mozanalysis.experiment import AnalysisWindow, TimeSeriesResult, TimeLimits
+
+@attr.s(frozen=True, slots=False)
+class HistoricalTarget:
+    """Query historical data.
+
+    The methods here query data in a way compatible with the following
+    principles, which are important for experiment analysis:
+
+    Example usage (in a colab notebook):
+
+        from google.colab import auth
+        auth.authenticate_user()
+        print('Authenticated')
+
+        from mozanalysis.sizing import HistoricalTarget
+        from mozanalysis.bq import BigQueryContext
+        from mozanalysis.metrics.desktop import active_hours, uri_count
+
+        bq_context = BigQueryContext(
+            dataset_id='your-dataset-id',  # e.g. mine's mbowerman
+            project_id='moz-fx-data-bq-data-science'  # this is the default anyway
+        )
+
+        ht = HistoricalTarget(
+            experiment_name='my_test_name',
+            start_date='2021-01-01',
+            num_dates_enrollment=8,
+            print_queries=True
+        )
+
+        # Run the query and get the results as a DataFrame
+        res = experiment.get_single_window_data(
```

Comment:
Should this be `res = ht.get_single_window_data(`?

Reply:
You're right, it should be. I'll fix that, thanks!

Reply:
Fixed!